lib/src/rules/provide_correct_intl_args.dart

```diff
@@ -0,0 +1,252 @@
+          methodInvocation,
+          parameterList,
+        );
+        break;
+    }
+  }
+
+  void _checkMessageMethod(
+      MethodInvocation methodInvocation, FormalParameterList parameterList) {
+    final argsArgument = methodInvocation.argumentList?.arguments
+        ?.whereType<NamedExpression>()
+        ?.where((argument) => argument.name.label.name == 'args')
+        ?.firstOrDefault()
+        ?.expression
+        ?.as<ListLiteral>();
+
+    final parameterSimpleIdentifiers = parameterList?.parameters
+            ?.map((parameter) => parameter.identifier)
+            ?.toList() ??
+        <SimpleIdentifier>[];
+
+    if (argsArgument != null && parameterSimpleIdentifiers.isEmpty) {
+      addIssue(_ArgsMustBeOmittedIssue(argsArgument));
+    }
+
+    if (argsArgument == null && parameterSimpleIdentifiers.isNotEmpty) {
+      addIssue(_NotExistArgsIssue(parameterList));
+    }
+
+    _checkAllArgsElementsMustBeSimpleIdentifier(argsArgument);
+
+    final argsSimpleIdentifiers =
+        argsArgument?.elements?.whereType<SimpleIdentifier>()?.toList() ??
+            <SimpleIdentifier>[];
+
+    _checkAllParametersMustBeContainsInArgs(
+        parameterSimpleIdentifiers, argsSimpleIdentifiers);
+    _checkAllArgsMustBeContainsInParameters(
+        argsSimpleIdentifiers, parameterSimpleIdentifiers);
+
+    final messageArgument = methodInvocation.argumentList?.arguments
+        ?.firstOrDefault()
+        ?.as<StringInterpolation>();
+
+    if (messageArgument != null) {
+      final interpolationExpressions = messageArgument.elements
+          .whereType<InterpolationExpression>()
+          .map((expression) => expression.expression)
+          .toList();
+
+      addIssues(interpolationExpressions
+          .where((item) => item is! SimpleIdentifier)
+          .map((item) => _MustBeSimpleIdentifierIssue(item)));
+
+      final interpolationExpressionSimpleIdentifiers =
+          interpolationExpressions.whereType<SimpleIdentifier>().toList();
+
+      if (argsArgument != null &&
+          parameterSimpleIdentifiers.isEmpty &&
+          interpolationExpressionSimpleIdentifiers.isEmpty) {
+        addIssue(_ArgsMustBeOmittedIssue(argsArgument));
+      }
+
+      if (interpolationExpressionSimpleIdentifiers.isNotEmpty) {
+        _checkAllInterpolationMustBeContainsInParameters(
+            interpolationExpressionSimpleIdentifiers,
+            parameterSimpleIdentifiers);
+        _checkAllInterpolationMustBeContainsInArgs(
+            interpolationExpressionSimpleIdentifiers, argsSimpleIdentifiers);
+      }
+    } else {
+      for (final item in parameterSimpleIdentifiers) {
+        addIssue(_ParameterMustBeOmittedIssue(item));
+        addIssue(_ArgsItemMustBeOmittedIssue(item));
+      }
+    }
+  }
+
+  void _checkAllArgsElementsMustBeSimpleIdentifier(
+      ListLiteral argsListLiteral) {
+    final argsElements = argsListLiteral?.elements ?? <CollectionElement>[];
+
+    addIssues(argsElements
```

Comment:
What do u think about uniting this logic and line 104 https://github.com/wrike/dart-code-metrics/pull/105/files#diff-edd15e45a7f4294a45ccba3691f1af08R114?

Reply:
for this case has been new method